presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -684,32 +693,40 @@ public void end()
         void afterValue();
     }
 
+    private interface BlockFieldConverter extends BlockConverter
```

Comment:
Most of converters can share the same interface, except ParquetListEntryConverter and ParquetMapEntryConverter. The field index doesn't apply to these two entry converters.